### ISSUE TITLE
Outline view

### ISSUE THREAD

**Issue by gantsevdenis** [helix-editor/helix#2425]
I went through documentation, and through features requests, but I couldn't find an "Outline" view like in VSCode for ex.:

![image](https://user-images.githubusercontent.com/25161465/167251697-c4c956a6-a75c-4703-ac62-5ed7fa498146.png)

It's quite difficult to navigate unknown code-base otherwise (in my opinion)

**Comment by kirawi** [comment]
I think this might be the LSP request to get a list of all the symbols: https://microsoft.github.io/language-server-protocol/specification#textDocument_documentSymbol

**Comment by n0s4** [comment]
FWIW there is a symbol picker with `space-s` but I guess you're looking for something more advanced?

**Comment by sudormrfbin** [comment]
LSP supports hierarchical symbols, but helix currently flattens then into a list and shows them with `space+s`. We could reuse the tree view that will soon be merged with the file tree (#2377) to display the symbol information in a tree interface.

**Comment by lukasjuhrich** [comment]
It may be worth noting that #2377 is closed and its successor #5768 is stalled and from the looks of it won't be merged.

Perhaps the question of a hierarchical symbol picker can be tackled separately? It seemed to me that the more broad question of „let's add a file picker“ has been received rather skeptically; perhaps people would be more willing to allow for this specific change when introduced on its own.

**Comment by psnelgrove-r7** [comment]
Found this issue while thinking about my workflow in large python projects in Neovim. Python LSP choices leave something to be desired when it comes to performance. But I found an alternative that I often find more useful and certainly more performant. That is to pull the outline directly from tree-sitter. 

The best plugin I found for this was [aerial.nvim](https://github.com/stevearc/aerial.nvim). In my option, Aerial does too many things with it's stateful outline buffer. Things like collapse/expand sections and code folding. That's what you'll see in the readme demo video.

However, what's most relevant to my comment is the document symbol picker. It's very similar to the Helix symbol picker with the candidates to the left and preview to the right. The two subtle but very nice UX improvement here are the (optionally rended if available) NerdFont and the actual tree-sitter node content next to the symbol for quick context.

![image](https://github.com/helix-editor/helix/assets/65517780/9b5ee627-7ed2-48f8-b59a-aebc54d62563)

I think it would be interesting if `space-s` could support the two different backends since I think the UX can be identical. 

To summarize, I think the main value propositions are:
1. Falling back to tree-sitter means that the default editor capabilities are there even if the LSP is not installed or configured.
2. A more performant per-file symbol picker for very large files.
3. More context on the line for each symbol pulled from the tree-sitter node (stating my ignorance here if the LSP response also gives this context and the picker just doesn't render it for simplicity).

**Comment by kirawi** [comment]
See https://github.com/helix-editor/helix/issues/1252 for that

**Comment by ClassAteam** [comment]
Um... Is this issue still relevant? It is for me.
